### PR TITLE
Adding nat gateway and aws router to instance merging to master

### PR DESCRIPTION
**Pull request by TheOnly-Co**
- Added aws_nat_gateway in vpc.tf.
- Added aws_router with router_table_id in instance.tf.

### CODE REVIEW

File: vpc.tf [hunk 1]
```diff
@@ -12,4 +12,13 @@ resource "aws_subnet" "core-infra" {
 
 resource "aws_internet_gateway" "gw"{
   vpc_id = aws_vpc.core-infra.id
+}
+
+resource "aws_nat_gateway" "gw"{
```
Comment (hunkeelin) [inline vpc.tf:17]: https://www.terraform.io/docs/providers/aws/r/internet_gateway.html

not nat gateway

File: instance.tf [hunk 2]
```diff
@@ -62,3 +62,13 @@ resource "aws_eip" "bastion-eip"{
 variable authorized_key {
     description = "The public ssh rsa key you generated"
 }
+
+resource "aws_route" "public_internet_gateway" {
+    route_table_id = "rtb-05fd94560d1c6e8c1"
```
Comment (hunkeelin) [inline instance.tf]: no hard code...... how many times i have to say this

File: instance.tf [hunk 3]
```diff
@@ -62,3 +62,13 @@ resource "aws_eip" "bastion-eip"{
 variable authorized_key {
     description = "The public ssh rsa key you generated"
 }
+
```
Comment (hunkeelin) [inline instance.tf:65]: Create your own route table. https://www.terraform.io/docs/providers/aws/r/route_table.html

File: vpc.tf (hunk 1 above)
Comment (TheOnly-Co) [reply vpc.tf:17]: you told me to create nat gateway and attach eip to it, so do i need the ```resource "aws_nat_gateway" "gw"{
  allocation_id = aws_instance.bastion-host.id
  subnet_id = aws_subnet.core-infra.id

  tags = {
    Name = "nat gateway"
  }
}```

**Pull request merged**